core/src/impl/Kokkos_Stacktrace.cpp

```diff
@@ -0,0 +1,248 @@
+  size_t cur = find_first_non_whitespace(s, 0);
+  while (cur != std::string::npos) {
+    const size_t end   = find_first_whitespace(s, cur);
+    const bool last    = (end == std::string::npos);
+    const size_t count = last ? end : size_t(end - cur);
+    c(s.substr(cur, count), last);
+    cur = find_first_non_whitespace(s, end);
+  }
+}
+
+// Search the whole backtrace, column by column, for "main".
+// This tells us what column has the function names.
+// While we're doing that, figure out the longest column,
+// so we can compute spacing correctly.
+
+std::tuple<bool, size_t, std::vector<size_t>> find_main_column(
+    const std::vector<std::string>& traceback) {
+  bool found_main = false;
+  size_t main_col = 0;
+  for (auto&& entry : traceback) {
+    size_t col_count = 0;
+    for_each_token(entry, [&](const std::string& s, bool) {
+      const size_t pos = s.find("main");
+      if (pos != std::string::npos) {
+        found_main = true;
+        main_col   = col_count;
+      }
+      ++col_count;
+    });
+    if (found_main) {
+      break;
+    }
+  }
+
+  // Make another pass to get the column lengths.
+  // Only demangle the column of functions.
+  std::vector<size_t> max_col_lens;
+  for (auto&& entry : traceback) {
+    size_t col_count = 0;
+    for_each_token(entry, [&](const std::string& s, bool) {
+      const size_t cur_col_len =
+          (found_main && col_count == main_col) ? demangle(s).size() : s.size();
+      ++col_count;
+      if (max_col_lens.size() < col_count) {
+        max_col_lens.push_back(cur_col_len);
+      } else {
+        const size_t old_max_len = max_col_lens[col_count - 1];
+        if (old_max_len < cur_col_len) {
+          max_col_lens[col_count - 1] = cur_col_len;
+        }
+      }
+    });
+  }
+  return std::make_tuple(found_main, main_col, max_col_lens);
+}
+
+void demangle_and_print_traceback_entry(
+    std::ostream& out, const std::string& traceback_entry,
+    const bool found_main, const size_t main_col,
+    const std::vector<size_t>& max_col_lens) {
+  std::vector<std::string> tokens;
+  size_t cur_col = 0;
+  for_each_token(traceback_entry, [&](const std::string& s, bool last) {
+    const size_t old_width(out.width());
+    out.width(max_col_lens[cur_col]);
+    try {
+      if (found_main && cur_col == main_col) {
+        out << demangle(s);
+      } else {
+        out << s;
+      }
+      if (!last) {
+        out << " ";
+      }
+      ++cur_col;
+    } catch (...) {
+      out.width(old_width);
+      throw;
+    }
+    out.width(old_width);
+  });
+}
+
+void demangle_and_print_traceback(std::ostream& out,
+                                  const std::vector<std::string>& traceback) {
+  const auto result = find_main_column(traceback);
+  for (auto&& entry : traceback) {
+    using std::get;
+    demangle_and_print_traceback_entry(out, entry, get<0>(result),
+                                       get<1>(result), get<2>(result));
```

Comment:
No reason to use ADL here IMHO; just use `std::get`

Reply:
This went away with using a struct for result.